cmd/template/nodepool/provider/aws.go

```diff
@@ -327,6 +328,10 @@ func newKubeadmConfigFromUnstructured(sshSSOPubKey string, sshdConfig string, o
 				Shell:  &shell,
 			},
 		}
+		kubeadmConfig.Spec.JoinConfiguration.NodeRegistration.KubeletExtraArgs = map[string]string{
+			"cloud-provider": "aws",
+			"node-labels":    fmt.Sprintf("giantswarm.io/machine-pool=%s", o.GetName()),
```

Comment:
Do we want to add this label into apiextensions, it might be we need to pick it up elsewhere?

Reply:
good call! Actually, it's already in there.